src/lower/lowerer_impl_imperative.cpp

```diff
@@ -1764,10 +1766,20 @@ Stmt LowererImplImperative::resolveCoordinate(std::vector<Iterator> mergers, ir:
   else {
     // Multiple position iterators so the smallest is the resolved coordinate
     if (emitVarDecl) {
-      return VarDecl::make(coordinate, Min::make(coordinates(mergers)));
+      // BEGIN GJ
+      return VarDecl::make(coordinate, Max::make(coordinates(mergers)));
```

Comment:
We need to take the `MAX` of the coordinates as the galloping target

Reply:
Agreed. Doing a hot fix for now; will properly fix after implementing scheduling.